### PR TITLE
Add `django_guid.middleware.guid_middleware` to `MIDDLEWARE` settings

### PR DESCRIPTION
**Pull request by jerabekjiri**
No-Issue

#### What is this PR doing:

`logging_cid` fails with `null` value. 

```
pulp_1   | django.db.utils.IntegrityError: null value in column "logging_cid" of relation "core_task" violates not-null constraint
pulp_1   | DETAIL:  Failing row contains (f9d4f0b4-16f0-47ff-b55b-db437b3c41cf, 2023-01-16 15:38:50.537753+00, 2023-01-16 15:38:50.537775+00, waiting, pulp_ansible.app.tasks.collections.sync, null, null, null, null, null, null, null, null, {"mirror": false, "optimize": true, "remote_pk": "5421af5a-5d70-..., {/api/automation-hub/pulp/api/v3/repositories/ansible/ansible/3d...)
```

#### How to reproduce: 
Dispatch any `sync, modify, sign` task (for example on endpoint: `/pulp/api/v3/repositories/ansible/ansible/<pk>/sync/`)

Issue appears on latest pulpcore `3.23.0` (Started happening after https://github.com/pulp/pulpcore/pull/3258) 

```
diff --git a/pulpcore/tasking/tasks.py b/pulpcore/tasking/tasks.py
index 4dc6c503a..dc16dc8a9 100644
--- a/pulpcore/tasking/tasks.py
+++ b/pulpcore/tasking/tasks.py
@@ -88,7 +88,7 @@ def dispatch(
     with transaction.atomic():
         task = Task.objects.create(
             state=TASK_STATES.WAITING,
-            logging_cid=(get_guid() or ""),
+            logging_cid=(get_guid()),
             task_group=task_group,
             name=func,
             args=args,
```

Pulp discussion: https://github.com/pulp/pulp_ansible/issues/1323 (PR: https://github.com/pulp/pulp_ansible/pull/1324)

**PR Author & Reviewers**: Keep or remove backport labels per [Backporting Guidelines](https://github.com/ansible/galaxy_ng/wiki/Backporting-Guidelines)
**Reviewers**: Look for sound code, no [code smells](https://www.codegrip.tech/productivity/everything-you-need-to-know-about-code-smells/), docs & test coverage
**Merger**: When merging, include the Jira issue link in the squashed commit

### CODE REVIEW

**Review by rochacbruno — APPROVED** [review]
LGTM


For reference: in future we will have a different way to contribute to middleware settings as you can see [here](https://github.com/dynaconf/dynaconf/issues/299#issuecomment-900616706)

**Comment by rochacbruno** [comment]
I am not sure why the insights check is not passing.

**Comment by jerabekjiri** [comment]
@rochacbruno I just noticed that in pulpcore `django_guid` is added to `INSTALLED_APPS`: https://github.com/pulp/pulpcore/blob/main/pulpcore/app/settings.py#L77, should we need to do that as well?

**Comment by rochacbruno** [comment]
> @rochacbruno I just noticed that in pulpcore `django_guid` is added to `INSTALLED_APPS`: https://github.com/pulp/pulpcore/blob/main/pulpcore/app/settings.py#L77
> 
> should we do that as well?

@jerabekjiri no need, we are merging with the INSTALLED_APPS so we get whatever is added to pulp https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/settings.py#L24

**Comment by awcrosby** [comment]
Does this PR end up populating a `logging_cid` field in the pulp_ansible endpoint? if so it can have a test added to pass the validation CI check.

**Review by awcrosby — CHANGES_REQUESTED** [review]
I do see http://localhost:5001/api/automation-hub/pulp/api/v3/tasks/ has `logging_cid`, so we can add an integration test to see that is not null.

**Comment by jerabekjiri** [comment]
@awcrosby I added the test to verify `logging_cid` val and `Correlation-ID` in headers

**Review by awcrosby — APPROVED** [review]
